cmd/opm/serve/serve.go

```diff
@@ -59,12 +68,32 @@ will not be reflected in the served content.
 	}
 
 	cmd.Flags().BoolVar(&s.debug, "debug", false, "enable debug logging")
+	cmd.Flags().StringVar(&s.profilePort, "profileport", "", "pprof HTTP endpoint port")
+	cmd.Flags().StringVar(&s.cpuProfile, "cpuprofile", "", "output file for CPU profiling data dump")
+	cmd.Flags().StringVar(&s.memProfile, "memprofile", "", "output file for memory profiling data dump")
 	cmd.Flags().StringVarP(&s.port, "port", "p", "50051", "port number to serve on")
 	cmd.Flags().StringVarP(&s.terminationLog, "termination-log", "t", "/dev/termination-log", "path to a container termination log file")
 	return cmd
 }
 
 func (s *serve) run(ctx context.Context) error {
+	if s.profilePort != "" {
+		go func() {
+			s.logger.Info(http.ListenAndServe("localhost:"+s.profilePort, nil))
```

Comment:
I saw this in the pprof docs but generally it's not recommended to start a goroutine without any way of stopping it. Maybe we could add a [WaitGroup](https://pkg.go.dev/sync#WaitGroup) with a value of one and then call `Done()` on it in the teardown logic below where the grpc server is torn down? 

Also, this logger would capture the server shutdown as an info-level log, which isn't a big deal but isn't great IMO. Could we swallow the shutdown error somehow?

Reply:
On discussion, we decided that this iteration didn't need to implement goroutine lifecycle management for pprof.  
There is no shutdown error, since it seems that the logger has been gc'd before the function returns. 

Restructured to output good diagnostics but no server shutdown error.

Reply:
SGTM